### ISSUE TITLE
Package arches with pyproject.toml?

### ISSUE THREAD

**Issue by jacobtylerwalls** [archesproject/arches#9941]
> Out of scope for this PR, but this might be an appetizing chance to move from `setup.py` to `pyproject.toml`, which is where python packaging seems to be coalescing. If in future versions we want to start shipping any other configuration, including tool configuration (some of which can _only_ be written in pyproject.toml, see how that got us into #9937 ...) we might wish we'd taken this opportunity to start new projects off on pyproject.toml from the get-go.

I'll open a new issue for this.

_Originally posted by @jacobtylerwalls in https://github.com/archesproject/arches/pull/9426#discussion_r1298505730_

**Comment by jacobtylerwalls** [comment]
Blocked by #10558
